embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java

```diff
@@ -340,20 +339,26 @@ public JdbcColumnOption get()
     }
 
     private boolean fetch(BatchSelect cursor,
-            List<ColumnGetter> getters, PageBuilder pageBuilder) throws SQLException
+            List<ColumnGetter> getters, PageBuilder pageBuilder, PluginTask task) throws SQLException
     {
         ResultSet result = cursor.fetch();
         if (result == null || !result.next()) {
             return false;
         }
 
+        Map<String, String> temporaryLastRecord = new HashMap<>();
+        Map<String, String> lastRecord = new HashMap<>();
+
         List<Column> columns = pageBuilder.getSchema().getColumns();
         long rows = 0;
         long reportRows = 500;
         do {
+            temporaryLastRecord.clear();
             for (int i=0; i < getters.size(); i++) {
                 int index = i + 1;  // JDBC column index begins from 1
                 getters.get(i).getAndSet(result, index, columns.get(i));
+                String k = columns.get(i).getName();
+                temporaryLastRecord.put(k, result.getString(index));
```

Comment:
Using Java's Map with String key is invalid because column names are case insensitive in SQL while String is case sensitive. If column names of the target table don't match with selecting column names, column names in the results may not match with selecting column names.